### PR TITLE
mongo interface with spring

### PR DESCRIPTION
**Pull request by Milihhard**

### CODE REVIEW

File: src/main/java/org/infobourg/betterrss/models/RssFlux.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+//@Document
+@Document(collection = "RssFlux")
+data class RssFlux(
+        @Id
+        var id: String="",
+        var idWorkspace: String="",
+        var idChannel: String="",
+        var idMessage: String="",
+        var idThread: String="",
```
Comment (IliasRedissi) [inline src/main/java/org/infobourg/betterrss/models/RssFlux.kt]: `idThread` should be nullable (`String?`):
- null when a feed is linked to a channel
- not-null when a feed is linked to a message
```suggestion
        var idThread: String? = null,
```

File: src/main/java/org/infobourg/betterrss/models/RssFlux.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package org.infobourg.betterrss.models
+
+import org.springframework.data.annotation.Id
+import org.springframework.data.mongodb.core.mapping.Document
+import java.time.LocalDateTime
+
+//@Document
+@Document(collection = "RssFlux")
+data class RssFlux(
```
Comment (IliasRedissi) [inline src/main/java/org/infobourg/betterrss/models/RssFlux.kt]: It's *RSS feed*. Need to change *Flux* to *Feed* in other places (repo, service).
```suggestion
@Document(collection = "RssFeed")
data class RssFeed(
```

File: src/main/java/org/infobourg/betterrss/models/RssFlux.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        @Id
+        var id: String="",
+        var idWorkspace: String="",
+        var idChannel: String="",
+        var idMessage: String="",
+        var idThread: String="",
+        var link: String="",
+        var lastUpdate: LocalDateTime= LocalDateTime.now()
```
Comment (IliasRedissi) [inline src/main/java/org/infobourg/betterrss/models/RssFlux.kt]: It's preferable to set immutable properties (`val`) for `data class` and uses `copy` to change properties.

Comment (Milihhard) [reply src/main/java/org/infobourg/betterrss/models/RssFlux.kt]: I think JPA uses mutation for creating their objects. putting it as a `var` is not for us but more for the framework. But if it's not the cas, we should indeed put it as a `val`

Comment (IliasRedissi) [reply src/main/java/org/infobourg/betterrss/models/RssFlux.kt]: I changed locally to `val` and it's still working.

File: src/main/java/org/infobourg/betterrss/models/RssFlux.kt (hunk 1 above)
Comment (Milihhard) [reply src/main/java/org/infobourg/betterrss/models/RssFlux.kt]: How a feed can be linked to a channel? can you do a thread from a channel and not a message?

**Pull request merged**